Feat/project form/create and edit form projects

andreajhl opened the pull request:
-Add get, post and put functions in projectServer.
-Add project creation and editing form, made with formik and ckeditor.
-add modularized function for rendering formik formulary errors
-add modularized function to obtain the url of the file type inputs
-You don't have styles yet because you have to decide it as a group.

Form:

-title: validation => long (minimum 4 letters)
-image: validation => must exist, and accept jpg or png, render the image above the input when loading it, as expected
-description: validation => must exist
-date: condition => does not allow dates less than the current day

![project](https://user-images.githubusercontent.com/75461111/139186248-8ce4760c-c68c-408f-9363-a86d0c9ff47a.png)

Commit:
feat: :sparkles: add form create and edit projects

Commit:
config initial

Commit:
feat: :sparkles: agrega funciones de post, get, put para projects

Commit:
feat: :sparkles: agrega funciones de post, get, put para projects

Commit:
feat: :sparkles: add request http (get,put,post) for compoent form projects

Commit:
feat: :sparkles: add form for edit and create projects

Commit:
style: :lipstick: modify layout

saralavanchy commented on src/Components/Projects/ProjectsForm.js:
```diff
@@ -1,32 +1,123 @@
-    console.log(initialValues);
-  }
+  var date = new Date();
+  date=date.toISOString().split('T')[0];
```
Im not sure what you are trying to do here but it looks like a good candidate for a functions on a date utils file

saralavanchy commented on src/Components/Projects/ProjectsForm.js:
```diff
@@ -1,32 +1,123 @@
-    e.preventDefault();
-    console.log(initialValues);
-  }
+  var date = new Date();
```
for what is this date? The name is too generic

andreajhl replied:
The data is so that the creation of the project has as a minimum date the current day

saralavanchy replied:
if its a minimumDate so it should be the name, if its the currentDate its should be the name, if its the projectDate it should be the name.

saralavanchy commented on src/Components/Projects/ProjectsForm.js:
```diff
@@ -1,32 +1,123 @@
-      setInitialValues({...initialValues, description: e.target.value})
-    }
-  }
+export default function ProjectsForm({idProyect}){
```
please, follow the convention of the proyect and use ```const ProjectsForn = ({projectId}) => {}```
Also its more redeable projectId than idProyect. In addition, in english is Project and not proyect

saralavanchy commented on src/Components/Projects/ProjectsForm.js:
```diff
@@ -1,32 +1,123 @@
-      <button className="submit-btn" type="submit">Send</button>
-    </form>
+  const [imgState,setImg] = useState('');
+  const [state, setstate] = useState('');
```
please use setState instead of setstate, and also, what this state is for, what do it represent? the name is too generic

saralavanchy commented on src/Components/Projects/ProjectsForm.js:
```diff
@@ -1,32 +1,123 @@
+    return (
+      <div className='form-container'>
+          <Formik
+              initialValues={state? {
```
this validation is too log to be done here, please create a function that get the initial values. Something like 
```
const getInitialValue= () => logic here 
```

saralavanchy commented on src/Components/Projects/ProjectsForm.js:
```diff
@@ -1,32 +1,123 @@
+                  image:'',
+                  up_date:''
+              }}
+              validate={(valores)=>{
```
Please create a validation function for this.

saralavanchy replied:
Also, if you are writting the code in english it should be values instead of valores

saralavanchy commented on src/Components/Projects/ProjectsForm.js:
```diff
@@ -1,32 +1,123 @@
+                  return errors;
+              }}
+              onSubmit={async (values)=>{
+                const stringreplace =/'<p>'|['</p>']/gi;
```
please create an onSubmit function to do this or a function whith a name that explains what is being done here.

saralavanchy commented on src/Components/Projects/ProjectsForm.js:
```diff
@@ -1,32 +1,123 @@
+              onSubmit={async (values)=>{
+                const stringreplace =/'<p>'|['</p>']/gi;
+                values.description=values.description.replace(stringreplace,'');
+                idProyect ?  putDataForm( idProyect, values ) :postDataForm(values)
```
please follow a linament of spaces if you use space before and after ?, use space before and after :

saralavanchy commented on src/Components/Projects/ProjectsForm.js:
```diff
@@ -1,32 +1,123 @@
+                            config={ {
+                                placeholder:"Write some activity description",
+                            } }
+                            onChange={( event, editor )=> {
```
create a function to do this

saralavanchy commented on src/Components/Projects/ProjectsForm.js:
```diff
@@ -1,32 +1,123 @@
+                            name="title"
+                            placeholder="Activity Title" 
+                          /> 
+                          <ErrorMessage name='title' component={()=><p className='form_div_error'>{errors.title}</p>}/>
```
this code is being repeted, please create a function to show it, something like this  (this code is only ilustrative of the suggested change):
```
const showErrorMessage = (name, message) =>  (<ErrorMessage name={name} component={()=><p className='form_div_error'>{message}</p>}/>)
```

saralavanchy commented on src/Components/Projects/ProjectsForm.js:
```diff
@@ -1,32 +1,123 @@
+                                accept="image/png, image/jpg" 
+                                id='image'                                
+                                name='image' 
+                                value={undefined}
```
Please, never use undefined as a value.

saralavanchy commented on src/Components/Projects/ProjectsForm.js:
```diff
@@ -1,32 +1,123 @@
+                                name='image' 
+                                value={undefined}
+                                onChange={(e) => {
+                                  setFieldValue("image",e)
```
please, move this code to a function

saralavanchy commented on src/Components/Projects/function/index.js:
```diff
@@ -0,0 +1,18 @@
+import axios from "axios";
```
this look to be a service, so it should be on the services folder on a file with the service name that you are creating. I think, (for the name of the folder and the endpoints) that its the projects service, so, create a ProjectsService with a getProject,  createProject and modifyProject functionallities

saralavanchy reviewed (changes requested):
I left some comments but the code was a little bit difficult to follow, so, after this changes I will maybe give a better feedback. Thanks!

Commit:
feat: :adhesive_bandage: modularization of the project form

it extracts in two functions the rendering of errors, and the obtaining of the url of the image inout

Commit:
feat: :sparkles: add request HTTP ( get, post, put) for the form project

Commit:
fix: :rewind: agg request http (get,post,put) for form project

Commit:
feat: :sparkles: add modularized functions to get the url of an image

add modularized functions to get the url of an image, and save it in a state

Commit:
feat: :sparkles: adds modularized functions to render form errors

adds modularized functions to render form errors made with formik, requires name and error

Commit:
delete console.log in disuse

Commit:
fix: :adhesive_bandage: fix saved image data

fix the saved image data, now saving the 64bit url

Commit:
fix: :adhesive_bandage: fix saved image data

fix the saved image data, now saving the 64bit url

Commit:
feat: :sparkles: create functions to validate formik form fields and return errors

It includes the replace function for the "<p></p>" of the ckditor component and a function to validate the values of the formik input

Commit:
feat: :sparkles: form modularization

extra the form and replace validations from ckeditor in a function in the common folder

saralavanchy committed:
pull request #2 autenticacion_de_usuarios/formulario_login

Email and password fields validation with Formik.

saralavanchy committed:
Revert "Formulario Login"

saralavanchy committed:
pull request #18 from alkemyTech/revert-2-feat/autenticacion_de_usuarios/formulario_login

Revert "Formulario Login"

saralavanchy commented on src/Components/Projects/ProjectsForm.js:
```diff
@@ -1,32 +1,110 @@
+              ...dataForm
+          }}
+          validate={(values)=>validateForm(values)}
+          onSubmit={async (values)=>{
```
en el return no debe haber functions, todo lo que sea codigo como por ejemplo esta functionallity onSubmit tiene que ir en una function que luego se llama aca, como hiciste con validateForm

saralavanchy commented on src/Components/Projects/ProjectsForm.js:
```diff
@@ -1,32 +1,110 @@
+          validate={(values)=>validateForm(values)}
+          onSubmit={async (values)=>{
+            const dataForm =replaceCKeditor(values)
+            projectId ?  modifiedProject( projectId, dataForm ) : createProject(dataForm);
```
modificar o crear proyecto es logica de negocios y debe estar en el service

saralavanchy commented on src/Components/common/validateForm.js:
```diff
@@ -0,0 +1,18 @@
+export const validateForm = (values) => {
```
si en el #13 creaste una carpeta para validations en common, aca este archivo deberia estar en esa carpeta tambien, siempre hay que intentar seguir una misma logica en todo el proyecto para tener el codigo de manera ordenada

saralavanchy commented on src/Components/common/messagesForm.js:
```diff
@@ -0,0 +1,11 @@
+
```
el nombre del file deberia ser ErrorsForm ya que lo que contiene es un ErrorsForm.

saralavanchy commented on src/Components/common/getUrlInputFile.js:
```diff
@@ -0,0 +1,12 @@
+
```
mismo feedback #13

saralavanchy commented on src/Components/Projects/ProjectsForm.js:
```diff
@@ -1,32 +1,110 @@
+  var minDateForm = new Date().toISOString().split('T')[0];
+
+  const [imgState,setImgState] = useState();
+  const [dataForm, setDataForm] = useState({
```
siempre es importante cuanto mayor claridad podamos dejar en el codigo mejor. si esta constante es llamada project es mas descriptiva de su contenido que dataForm. dataForm es muy generica y puede ser cualquier cosa que sea data de un form.

Commit:
feat: :fire: delete unused files
the functions in these files were moved to new files in their corresponding folders

Commit:
feat: :sparkles: add function to render form errors made with formik

Commit:
feat: :sparkles: add function to read and set url of input file
add function to read and set its value through a callback function

Commit:
feat: :sparkles: add function to remove the "<p> </p>" from the value of ckeditor

Commit:
feat: :sparkles: add function to render form errors made with formik

Commit:
feat: :sparkles: modify variable names and separate functions in external folders
modify variable names to be more representative and extract functions in the common folder to make the component modularized and reusable

Commit:
feat: :sparkles: add route for edit projects in file APP

Commit:
feat: :sparkles: modified name for validations form projects

modified "name => title"

Commit:
feat: :sparkles: modifies imports of functions and agg obtaining id by params

Commit:
feat: :sparkles: add new function to dispatch creation or modification and modified name for route

andreajhl committed:
Merge branch 'development' into feat/project-Form/create-and-edit-form-projects

Commit:
feat: :sparkles: modified up_date a due_date

Commit:
Merge branch feat/project-Form/create-and-edit-form-projects of https://github.com/alkemyTech/OT91-server into feat/project-Form/create-and-edit-form-projects

Commit:
feat: :sparkles: modified route request api rest

Commit:
feat: :fire: remove import and delete file unused to counter

Commit:
Merge branch development into feat/project-Form/create-and-edit-form-projects

andreajhl committed:
Merge branch 'development' into feat/project-Form/create-and-edit-form-projects
